### PR TITLE
Attributes: Warn and fill .toggleClass( boolean )

### LINKED ISSUE #175: Warn and fill `.toggleClass()` and `.toggleClass( boolean )`
https://github.com/jquery/jquery/issues/2491

https://github.com/jquery/jquery/commit/a4d16a26abe3ea36337f7bfc4d1bfe13b893cd17

### PR DESCRIPTION
**Pull request by dmethvin**
Fixes #175

### CODE REVIEW

**Comment by dmethvin** [comment]
Note to self: Need to update the upgrade guide, it currently says only the no-args form is deprecated but the single-boolean form is deprecated as well.

File: test/attributes.js [hunk 1]
```diff
@@ -41,3 +41,40 @@ QUnit.test( ".removeAttr( boolean attribute )", function( assert ) {
...
+		assert.equal( e.attr( "class" ), "classy", "class was toggle-set" );
+		e.toggleClass( "classy", false );
+		assert.equal( e.attr( "class" ), "", "class was toggle-removed" );
+	} );
+
+	expectWarning( "toggleClass full-className manipulations", function() {
+		e.addClass( "testD testE" );
+		assert.ok( e.is( ".testD.testE" ), "Assert class present" );
+		e.toggleClass();
+		assert.ok( !e.is( ".testD.testE" ), "Assert class not present" );
```
Comment (mgol) [inline test/attributes.js]: Maybe test that `e` has none of those classes and not that it just doesn't have both at the same time?

File: test/attributes.js [hunk 2]
```diff
@@ -41,3 +41,40 @@ QUnit.test( ".removeAttr( boolean attribute )", function( assert ) {
...
+
+	expectNoWarning( "toggleClass not full className", function() {
+		e.attr( "class", "" );
+		e.toggleClass( "classy" );
+		assert.equal( e.attr( "class" ), "classy", "class was toggle-set" );
+		e.toggleClass( "classy", false );
+		assert.equal( e.attr( "class" ), "", "class was toggle-removed" );
+	} );
+
+	expectWarning( "toggleClass full-className manipulations", function() {
```
Comment (mgol) [inline test/attributes.js]: If only the single boolean arg or no arg case was issuing errors this test would still pass. Maybe assert the number of warnings as well?

File: warnings.md [hunk 3]
```diff
@@ -166,3 +166,9 @@ See jQuery-ui [commit](https://github.com/jquery/jquery-ui/commit/c0093b599fcd58
 **Cause:** The standard way to add new custom selectors through jQuery is `jQuery.expr.pseudos`. These two other aliases are deprecated, although they still work as of jQuery 3.0.
 
 **Solution:** Rename any of the older usage to `jQuery.expr.pseudos`. The functionality is identical.
+
+### JQMIGRATE: jQuery.fn.toggleClass( boolean ) is deprecated
```
Comment (mgol) [inline warnings.md]: `jQuery.fn.toggleClass( boolean )` and `jQuery.fn.toggleClass()`

File: test/attributes.js (hunk 1 above)
Comment (dmethvin) [inline test/attributes.js]: These tests were copied from core pretty much as-is so I hesitate to add more assertions. Whatever it was doing before was lightly documented and we're deprecating it now.

File: test/attributes.js (hunk 2 above)
Comment (dmethvin) [inline test/attributes.js]: Yeah I'll tighten that up.

File: warnings.md (hunk 3 above)
Comment (dmethvin) [inline warnings.md]: How about `[ boolean ]` to keep the error message short? The longer description already covers both cases.

Comment (mgol) [inline warnings.md]: 👍

File: test/attributes.js (hunk 1 above)
Comment (mgol) [inline test/attributes.js]: OK then!

**Pull request closed without merging**